### ISSUE TITLE
[bug] digest validation only happening when specified

### ISSUE THREAD

**Issue by vbatts** [vbatts/go-mtree#64]
```
[vbatts@bananaboat] {master *} ~/containers$ gomtree -c -T ./fedora.tar -K sha512digest,nlink > fedora.mtree.tar
[vbatts@bananaboat] {master *} ~/containers$ gomtree -f ./fedor.tara.mtree -p ./fedora/
[vbatts@bananaboat] {master *} ~/containers$ echo "" >> fedora/etc/fstab 
[vbatts@bananaboat] {master *} ~/containers$ gomtree -f ./fedora.mtree.tar -p ./fedora/
"etc/fstab": keyword "size": expected 313; got 314
[vbatts@bananaboat] {master *} ~/containers$ gomtree -f ./fedora.mtree.tar -p ./fedora/ -k sha512digest
"etc/fstab": keyword "sha512digest": expected 84e7b670c534f3f7e595c5ad5825de555d0d09cd37fbdb94beeee1023d9a1b5ceeae75ae4d31d9d9a974017894aed7f0ca9e7d66d34cbc8c84efb19b264b8e54; got 0df9ed341b0fceecceadc91e54bff16559512a2a27a6988c5111d92f1004d4c28eae90f70ba4565563615cf45d3bab57a15f0546cdc3a7ca02a6f8c6257a2f3b
```

The `sha512digest` keyword is in the manifest, but only being checked against when specifically instructed to do so.

**Comment by stephen679** [comment]
need to make sure all keywords found in a manifest are evaluated, unless the caller uses the `-k` flag option to reduce keywords to check

**Comment by cyphar** [comment]
This is something I was considering while writing in #48. The other issue is if we have `-k` set to something which doesn't already exist in the manifest. IMO we should be defaulting to the set of keywords in the spec rather than the `Default{Tar,}Keywords`.

**Comment by vbatts** [comment]
fair. Perhaps we can use the logic from the `-list-keywords` flag to detect
keywords first.

**Comment by cyphar** [comment]
I've done _something_ like that in #48: https://github.com/vbatts/go-mtree/pull/48/files#diff-e8b2d04ec2884011801582c86d040d36R188. But I can make a new PR where we default to `CollectUsedKeywords` and then make modifications to the default as per usual?

**Comment by vbatts** [comment]
This was fixed by #92
